upload/modules/Store/gateways/PayPalBusiness/gateway.php

```diff
@@ -29,34 +29,71 @@ public function processOrder(Order $order): void {
             return;
         }
 
-        $payer = new \PayPal\Api\Payer();
-        $payer->setPaymentMethod('paypal');
+        if (!$order->isSubscriptionMode()) {
+            // Single payment
+            $payer = new \PayPal\Api\Payer();
+            $payer->setPaymentMethod('paypal');
 
-        $amount = new \PayPal\Api\Amount();
-        $amount->setTotal(Store::fromCents($order->getAmount()->getTotalCents()));
-        $amount->setCurrency($order->getAmount()->getCurrency());
+            $amount = new \PayPal\Api\Amount();
+            $amount->setTotal(Store::fromCents($order->getAmount()->getTotalCents()));
+            $amount->setCurrency($order->getAmount()->getCurrency());
 
-        $transaction = new \PayPal\Api\Transaction();
-        $transaction->setAmount($amount);
-        $transaction->setDescription($order->getDescription());
-        $transaction->setInvoiceNumber($order->data()->id);
+            $transaction = new \PayPal\Api\Transaction();
+            $transaction->setAmount($amount);
+            $transaction->setDescription($order->getDescription());
+            $transaction->setInvoiceNumber($order->data()->id);
+            $transaction->setPurchaseOrder($order->data()->id);
+            $transaction->setCustom($order->data()->id);
 
-        $redirectUrls = new \PayPal\Api\RedirectUrls();
-        $redirectUrls->setReturnUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=success'))
-            ->setCancelUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=cancel'));
+            $redirectUrls = new \PayPal\Api\RedirectUrls();
+            $redirectUrls->setReturnUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=success'))
+                ->setCancelUrl(rtrim(URL::getSelfURL(), '/') . URL::build('/store/process/', 'gateway=PayPalBusiness&do=cancel'));
 
-        $payment = new \PayPal\Api\Payment();
-        $payment->setIntent('sale')
-            ->setPayer($payer)
-            ->setTransactions([$transaction])
-            ->setRedirectUrls($redirectUrls);
+            $payment = new \PayPal\Api\Payment();
+            $payment->setIntent('sale')
+                ->setPayer($payer)
+                ->setTransactions([$transaction])
+                ->setRedirectUrls($redirectUrls);
 
-        try {
-            $payment->create($apiContext);
+            try {
+                $payment->create($apiContext);
+
+                Redirect::to($payment->getApprovalLink());
+            } catch (\PayPal\Exception\PayPalConnectionException $ex) {
+                ErrorHandler::logCustomError($ex->getData());
+            }
+        } else {
+            // Payment subscription
+            $product = null;
+            foreach ($order->getProducts() as $item) {
+                $product = $item;
+                break;
+            }
```

Comment:
What's the idea behind this? Isn't the products list meant to be a single value in this case? Just use $order->getProducts()[0] then?

Reply:
Can't do 0 as the key is the product id

Reply:
Then use something like:
```php
$product = array_values($order-getProducts())[0]
```
or 
```php
$product = reset($order-getProducts())
```